pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -292,8 +293,51 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 		return controllerruntime.Result{}, errors.Wrap(err, "RoleBinding reconcile failed")
 	}
 
-	containerEnvs := r.populateContainerEnvs(ctx, rs, reconcilerRef.Name)
-	mut := r.mutationsFor(ctx, rs, containerEnvs)
+	notificationEnabled, err := util.NotificationEnabled(ctx, r.client, rs.Namespace, rs.Annotations, rs.Spec.NotificationConfig)
+	if err != nil {
+		log.Error(err, "Check notification subscription failed")
+		reposync.SetStalled(rs, "Notification", err)
+		// check notification subscription errors should always trigger retry (return error),
+		// even if status update is successful.
+		_, updateErr := r.updateStatus(ctx, currentRS, rs)
+		if updateErr != nil {
+			log.Error(updateErr, "Object status update failed",
+				logFieldObject, rsRef.String(),
+				logFieldKind, r.syncKind)
+		}
+		// Use the upsert error for metric tagging.
+		metrics.RecordReconcileDuration(ctx, metrics.StatusTagKey(err), start)
+		return controllerruntime.Result{}, errors.Wrap(err, "Check notification subscription failed")
+	}
+
+	if notificationEnabled && rs.Spec.NotificationConfig != nil &&
```

Comment:
It only validates if the Secret exists. We should also validate if the ConfigMap exists.

We also need to validate when it has the subscribe annotation, but NotificationConfig is nil. It should return an error.

We can probably do the validation inside the `NotificationEnabled` function.

Reply:
I moved all of the validation into the `NotificationEnabled` function and added more checks, PTAL